coderd/database/dbtestutil/driver.go

```diff
@@ -0,0 +1,79 @@
+package dbtestutil
+
+import (
+	"context"
+	"database/sql/driver"
+
+	"github.com/lib/pq"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database"
+)
+
+var _ database.DialerConnector = &Connector{}
+
+type Connector struct {
+	name   string
+	driver *Driver
+	dialer pq.Dialer
+}
+
+func (c *Connector) Connect(_ context.Context) (driver.Conn, error) {
+	if c.dialer != nil {
+		conn, err := pq.DialOpen(c.dialer, c.name)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to dial open connection: %w", err)
+		}
+
+		c.driver.Connections <- conn
+
+		return conn, nil
+	}
+
+	conn, err := c.driver.Open(c.name)
```

Comment:
This is circular.  `Driver.Open()` creates a new `Connector` and calls into it.  Fortunately, the pubsub creates a dialer, but anyone else who uses this might not and will overflow their stack.

You need to use `pq.Driver{}` directly here.